### PR TITLE
feat: add inline type hints

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

File: google/cloud/firestore_v1/__init__.py [hunk 1]
```diff
@@ -97,6 +97,13 @@
 # from .types.write import DocumentDelete
 # from .types.write import DocumentRemove
 from .types.write import DocumentTransform
+from typing import List
+
+__all__: List[str]
+__version__: str
+
+__all__: List[str]
+__version__: str
```
Comment (rafilong) [inline google/cloud/firestore_v1/__init__.py:106]: Is there a reason for this duplication?

File: google/cloud/firestore_v1/async_batch.py [hunk 2]
```diff
@@ -30,10 +31,10 @@ class AsyncWriteBatch(BaseWriteBatch):
             The client that created this batch.
     """
 
-    def __init__(self, client):
+    def __init__(self, client) -> None:
         super(AsyncWriteBatch, self).__init__(client=client)
 
-    async def commit(self):
+    async def commit(self) -> Union[Coroutine[Any, Any, list], list]:
```
Comment (rafilong) [inline google/cloud/firestore_v1/async_batch.py:37]: I'm a bit lost - why is the `Union` necessary?

File: google/cloud/firestore_v1/base_query.py [hunk 3]
```diff
@@ -171,7 +186,7 @@ def _client(self):
         """
         return self._parent._client
 
-    def select(self, field_paths):
+    def select(self, field_paths) -> "BaseQuery":
```
Comment (rafilong) [inline google/cloud/firestore_v1/base_query.py:189]: What is the significance of the quotes around `"BaseQuery"`?

File: google/cloud/firestore_v1/types/__init__.py [hunk 4]
```diff
@@ -68,6 +68,99 @@
...
+    str,
+    str,
+    str,
+    str,
+    str,
+    str,
+    str,
+    str,
+    str,
+]
```
Comment (rafilong) [inline google/cloud/firestore_v1/types/__init__.py:163]: Similarly confused about the duplication here.

File: google/cloud/firestore_v1/types/common.py [hunk 5]
```diff
@@ -19,6 +19,11 @@
 
 
 from google.protobuf import timestamp_pb2 as timestamp  # type: ignore
+from typing import Any
+
+__protobuf__: Any
+
+__protobuf__: Any
```
Comment (rafilong) [inline google/cloud/firestore_v1/types/common.py:26]: Once again duplication question.

File: google/cloud/firestore_v1/types/write.py [hunk 6]
```diff
@@ -21,6 +21,12 @@
 from google.cloud.firestore_v1.types import common
 from google.cloud.firestore_v1.types import document as gf_document
 from google.protobuf import timestamp_pb2 as timestamp  # type: ignore
+from typing import Any
+
+__protobuf__: Any
+__protobuf__: Any
+
+__protobuf__: Any
```
Comment (rafilong) [inline google/cloud/firestore_v1/types/write.py:29]: Why three copies?

File: google/cloud/firestore_v1/types/query.py [hunk 7]
```diff
@@ -20,6 +20,11 @@
 
 from google.cloud.firestore_v1.types import document
 from google.protobuf import wrappers_pb2 as wrappers  # type: ignore
+from typing import Any
+
+__protobuf__: Any
+
+__protobuf__: Any
```
Comment (rafilong) [inline google/cloud/firestore_v1/types/query.py:27]: Duplication?

File: google/cloud/firestore_v1/types/firestore.py [hunk 8]
```diff
@@ -24,6 +24,11 @@
 from google.cloud.firestore_v1.types import write
 from google.protobuf import timestamp_pb2 as timestamp  # type: ignore
 from google.rpc import status_pb2 as gr_status  # type: ignore
+from typing import Any
+
+__protobuf__: Any
+
+__protobuf__: Any
```
Comment (rafilong) [inline google/cloud/firestore_v1/types/firestore.py:31]: Duplication?

File: google/cloud/firestore_v1/types/document.py [hunk 9]
```diff
@@ -21,6 +21,11 @@
 from google.protobuf import struct_pb2 as struct  # type: ignore
 from google.protobuf import timestamp_pb2 as timestamp  # type: ignore
 from google.type import latlng_pb2 as latlng  # type: ignore
+from typing import Any
+
+__protobuf__: Any
+
+__protobuf__: Any
```
Comment (rafilong) [inline google/cloud/firestore_v1/types/document.py:28]: Duplication? Sorry about duplicate comments if this is benign.

File: google/cloud/firestore_v1/__init__.py (hunk 1 above)
Comment (crwilcox) [reply google/cloud/firestore_v1/__init__.py:106]: nope. This is likely a bug from the generation step of adding these. will remove this one. do flag others. I found I think most, but some may have sneaked by.

File: google/cloud/firestore_v1/base_query.py (hunk 3 above)
Comment (crwilcox) [reply google/cloud/firestore_v1/base_query.py:189]: you can't specify the self type as far as I can tell other ways

File: google/cloud/firestore_v1/__init__.py (hunk 1 above)
Comment (rafilong) [reply google/cloud/firestore_v1/__init__.py]: I believe I flagged all duplicates in my initial review. I was actually worried about duplicate comments about duplication, so I'm happy that that wasn't an issue :)

File: google/cloud/firestore_v1/async_batch.py (hunk 2 above)
Comment (crwilcox) [reply google/cloud/firestore_v1/async_batch.py:37]: This should have been removed. it turns out the actual type here is list. the coroutine bit was added errantly by pytype generation.

File: noxfile.py [hunk 10]
```diff
@@ -22,7 +22,7 @@
 
 import nox
 
-
+PYTYPE_VERSION = "pytype==2020.7.24"
```
Comment (tseaver) [inline noxfile.py:25]: This change (and below) gets wiped out by `synthtool`.

File: setup.cfg [hunk 11]
```diff
@@ -17,3 +17,14 @@
 # Generated by synthtool. DO NOT EDIT!
 [bdist_wheel]
 universal = 1
+
+[pytype]
```
Comment (tseaver) [inline setup.cfg:21]: Likewise wiped out by `synthtool`.

File: google/cloud/firestore_v1/types/__init__.py [hunk 12]
```diff
@@ -68,6 +68,54 @@
     BatchWriteRequest,
     BatchWriteResponse,
 )
+from typing import Tuple
+
+
+__all__: Tuple[
```
Comment (tseaver) [inline google/cloud/firestore_v1/types/__init__.py:74]: Why a tuple?  It would normally be `List[str]`?

**Pull request merged**